src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/pipeline_management/create_or_update_rule.py

```diff
@@ -58,6 +60,40 @@ def lambda_handler(event, _):
         .get("properties", {})
         .get("account_id")
     )
+
+    # Resolve codecommit source_account_id in case it is not set
+    if source_provider == "codecommit" and not source_account_id:
+        # Evaluate as follows:
+        # If not set, we have to set it with
+        #   - default_scm_codecommit_account_id (if exists)
+        #   - or ADF_DEPLOYMENT_ACCOUNT_ID
+        # If set, we are done anyways
+        LOGGER.debug(
+            "source_account_id not found in source_props - ADF will set "
+            "it from SSM param default_scm_codecommit_account_id.",
+        )
+        deployment_account_id = DEPLOYMENT_ACCOUNT_ID
+        try:
+            parameter_store = ParameterStore(DEPLOYMENT_ACCOUNT_REGION, boto3)
+            default_scm_codecommit_account_id = parameter_store.fetch_parameter(
+                "/adf/scm/default-scm-codecommit-account-id",
+            )
+        except ParameterNotFoundError:
+            default_scm_codecommit_account_id = deployment_account_id
+            LOGGER.debug(
+                "default_scm_codecommit_account_id not found in SSM - "
+                "Fall back to deployment_account_id.",
+            )
+        source_account_id = default_scm_codecommit_account_id
+
+        # Create the properties object if it does not exist
+        if not pipeline["default_providers"]["source"].get("properties") is None:
```

Comment:
I think there is a small merge error here. It states `if not` and `source.properties is None`. In other words, this will evaluate to True if it is set. While the result creates a new properties dictionary.

```suggestion
        if pipeline["default_providers"]["source"].get("properties") is None:
```

Reply:
I agree. Changed now